src/execution/execute-afm.ts

```diff
@@ -114,17 +116,18 @@ export class ExecuteAfmModule {
     }
 
     private getOnePage(pollingUri: string, offset: number[], prevResult: any = null): any {
-        return this.fetchExecutionResult(pollingUri, offset).then((executionResult: Execution.IExecutionResult) => {
-            if (executionResult === null) {
-                return null;
-            }
-
-            const newResult = prevResult ? mergePageData(prevResult, executionResult) : executionResult;
-
-            const nextOffset = nextPageOffset(executionResult.executionResult.paging);
-            return nextOffset
-                ? this.getOnePage(pollingUri, nextOffset, newResult)
-                : newResult;
-        });
+        return this.fetchExecutionResult(pollingUri, offset)
+            .then((executionResult: Execution.IExecutionResultWrapper) => {
+                if (executionResult === null) {
+                    return null;
+                }
+
+                const newResult = prevResult ? mergePageData(prevResult, executionResult) : executionResult;
+
+                const nextOffset = nextPageOffset(executionResult.executionResult.paging);
```

Comment:
I would rather rename variable to executionResultWrapper and similar for other variables which contains IExecutionResultWrapper 

`executionResultWrapper.executionResult.paging`